packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,572 @@
+        <SideNavItemLabelContainer
+          odysseyDesignTokens={odysseyDesignTokens}
+          isIconVisible={!!startIcon}
+          isDisabled={isDisabled}
+        >
+          {label}
+          {severity && (
+            <Box
+              sx={{
+                marginLeft: odysseyDesignTokens.Spacing2,
+              }}
+            >
+              <Status severity={severity} label={statusLabel || ""} />
+            </Box>
+          )}
+        </SideNavItemLabelContainer>
+        {endIcon && endIcon}
+      </Link>
+    </SideNavListItemContainer>
+  );
+};
+
+export type SideNavProps = {
+  /**
+   * Side Nav header text that is usually reserved to show the App name
+   */
+  navHeaderText: string;
+  /**
+   *  Determines whether the side nav is collapsible
+   */
+  isCollapsible?: boolean;
+  /**
+   *  Triggers when the side nav is collapsed
+   */
+  onCollapse?(): void;
+  /**
+   * Nav items in the side nav
+   */
+  sideNavItems: SideNavItem[];
+  /**
+   * Footer items in the side nav
+   */
+  footerItems?: SideNavFooterItem[];
+} & Pick<HtmlProps, "testId">;
+
+const SideNav = ({
+  navHeaderText,
+  isCollapsible,
+  onCollapse,
+  sideNavItems,
+  footerItems,
+}: SideNavProps) => {
+  const [sideNavCollapsed, setSideNavCollapsed] = useState(false);
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const processedSideNavItems = useMemo(
+    () =>
+      sideNavItems.map((item) => ({
+        ...item,
+        children: item.children?.map((childProps) => (
+          <SideNavItemContent key={childProps.id} {...childProps} />
+        )),
+      })),
+    [sideNavItems],
+  );
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        height: "100vh",
+      }}
+    >
+      <SideNavCollapsedContainer
+        tabIndex={0}
+        odysseyDesignTokens={odysseyDesignTokens}
+        sideNavCollapsed={sideNavCollapsed}
+        onClick={() => setSideNavCollapsed(!sideNavCollapsed)}
+        onKeyDown={(event) => {
+          event.key === "Enter" && setSideNavCollapsed(!sideNavCollapsed);
+        }}
+      >
+        <ExpandLeftIcon
+          sx={{
+            margin: `0 ${odysseyDesignTokens.Spacing1}`,
+          }}
+        ></ExpandLeftIcon>
+      </SideNavCollapsedContainer>
+      <SideNavContainer
+        odysseyDesignTokens={odysseyDesignTokens}
+        sideNavCollapsed={sideNavCollapsed}
+      >
+        <Box
+          sx={{
+            position: "sticky",
+            top: 0,
+          }}
+        >
+          <SideNavHeader
+            navHeaderText={navHeaderText}
+            isCollapsible={isCollapsible}
+            onCollapse={() => {
+              setSideNavCollapsed(!sideNavCollapsed);
+              onCollapse && onCollapse();
+            }}
+          />
+        </Box>
+        <Box
+          sx={{
+            flex: 1,
+            overflowY: "auto",
+          }}
+        >
+          <SideNavListContainer>
+            {processedSideNavItems?.map((item) => {
```

Comment:
This is a lot of logic to keep in the component return. Can any of these be pulled out as a memo or callback?

Reply:
I did move as much as I can and left the component rendering logic inside